### PR TITLE
Add missing label to border radius range control.

### LINKED ISSUE #42098: Radius control: the range input is unlabelled
### Description

In the Radius control, the slider input (type=range) is unlabelled. All form controls need to be labelled.

### Step-by-step reproduction instructions

- edit a block that supports the radius control e.g.: Columns
- make sure the Radius control is enabled in the 'Dimensions 
- use the Chrome dev tools to inspect the source of the Radius control that is displayed in the sidebar
- select the radius slider input (type=range)
- observe there's no associated label element or any aria-label / aria-labelledby
- in the Chrome dev tools, go to the Accessibility tab
- observe the accessible 'Name' under the computed properties is empty:

Screenshot 2022-07-01 at 16 41 49

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by afercia**
Fixes https://github.com/WordPress/gutenberg/issues/42098

## What?

Adds a missing label to the range input in the Border radius control.

## Why?

All form controls need to be labelled.

## How?

- adds a `label` prop with meaningful value
- adds `hideLabelFromVision` prop to make the label visually hidden
- updates docs and story of the range control: the `hideLabelFromVision` was added in a previous PR but it was undocumented

## Testing Instructions
- Edit a post and select a block that uses a Border radius control e.g. the Columns block.
- Check the Border radius control in the inspector sidebar.
- Check that:
  - the range input has a visually hidden `<label>` element associated with a `for` attribute to its ID
  - the range input has an `aria-label` attribute with the same text of the `<label>` element

Optionally:
- launch the story book: `npm run storybook:dev`
- check the Range control component
- check that toggling the `hideLabelFromVision` prop toggles the label visibility

Note: using both a `<label>` element and an `aria-label` attribute is redundant. The current component just works this way and a refactoring is out of the scope of this PR.

### CODE REVIEW

**Review by ciampo — COMMENTED** [review]
Thank you for your contribution, @afercia !

The improvements to the `RangeControl` component that you're proposing are actually going to be added by #40535 and #41444 (very soon, hopefully in a couple of days!).

If you don't mind, I'd rather keep this PR focused on making changes only to `BorderRadiusControl` — would you mind removing the changes to `RangeControl` ?


Otherwise, with respect to the changes to `BorderRadiusControl`, I have a few questions:

- it looks like the `aria-label` may be enough to label this input field ([source](https://www.w3.org/WAI/tutorials/forms/labels/))
- in case it's not enough, I wonder if we should also:
  - look into adding a `label` to the `input[type='number]` rendered alongside the slider input
  - look into adding the same labels to the four `input[type='number]` that are rendered when the border radii are un-linked


cc'ing @alexstine in case he has any inputs on the `label` vs `aria-label` attributes, and @aaronrobertshaw (since he has experience working on both `BorderRadiusControl` and `RangeControl`)

**Comment by alexstine** [comment]
It is always better to have the HTML label. However, the aria-label works when providing the HTML label is simply not practical and the amount of work to add it would be too much of an undertaking.

TLDR: Use the aria-label only when HTML label takes 2X+ the work.

**Comment by aaronrobertshaw** [comment]
> in case he has any inputs on the label vs aria-label attributes, and @aaronrobertshaw (since he has experience working on both BorderRadiusControl and RangeControl)

Thanks for the ping.

As this PR shows, there's not much involved in getting an HTML label in place and still hidden from view for the range input.

Is there any value in having all the approaches to labelling within the component consistent? That is, the `UnitControl` in the linked view only relies on an `aria-label` attribute. So does the `input[type='number']` for the `RangeControl` if it has a `label` set.

> look into adding the same labels to the four input[type='number] that are rendered when the border radii are un-linked

The `UnitControl` components when the border radii are unlinked currently get aria-labels describing which corner they represent e.g. `Top left`, `Bottom right` etc.

> adds a label prop with meaningful value

Would another option to getting a meaningful HTML label in place be to replace the `legend` used in the component?

> If you don't mind, I'd rather keep this PR focused on making changes only to BorderRadiusControl — would you mind removing the changes to RangeControl ?

Narrowing the scope of this PR to exclude the RangeControl changes that are already covered elsewhere sounds good to me.

**Comment by ciampo** [comment]
> Is there any value in having all the approaches to labelling within the component consistent? That is, the UnitControl in the linked view only relies on an aria-label attribute. So does the input[type='number'] for the RangeControl if it has a label set.

That is the main question that I'm asking — I don't have any issues with the addition of a `label` to `RangeControl` — but, as you noted, every other `input` rendered by `BorderRadiusControl` would still be labels through `aria-label` instead.

**Comment by afercia** [comment]
Removed the changes to `RangeControl`.

Re: best practices for labelling, we should keep into consideration a wide range of user needs and not focus only on screen reader users needs. Worth also mentioning that screen readers are used also by low vision users. This PR isn't the best place for detailed explanations 🙂 Very shortly:

**A visible `<label>` element associated with for/id attributes is always preferable**
It helps all users. It's also clickable and helps targeting the associated form element. Think at small input fields, checkboxes, radio buttons, etc.

**When hiding labels**
Whether it's done with a visually hidden `<label>` element or an `aria-label` attribute or other techniques, it's always a trade-off for accessibility. For example, an aria-label works for blind screen reader users but it doesn't for other users. Think at speech recognition software users: they need to know the name of a form element to voice a command like 'Click border radius'. That's the reason why we use tooltips: to visually expose the name of a form element or button. Users would have to focus the element to make the tooltip appear, and that's a big compromise, but at least they have a chance to learn the name of the form control. All the input controls should use tooltips for this reason. When unlinked, the input names `Top left', Top right', etc. do appear within tooltips. When linked, they don't. 
In short, the name of a form element should always be visually exposed in some way.

**Note about the usage of `<fieldset>` plus `<legend>`**
I like how they're used in the border radius control. The `<fieldset>` element groups together logically related form elements. The `<legend>` gives the group a name. The way screen readers announce the group name varies: some screen readers announce it when entering the group, some screen readers even repeat the group name when tabbing into each form element. This allows, for example, to clarify what 'Top left' refers to (when the sides are unlinked). The screen reader will announce something like:

'**Top left**, incrementable edit text number field, **Radius**'

Without the legend, it would only announce:

'**Top left**, incrementable edit text number field'

Users wouldn't have a clue what 'Top left' is about and they would be forced to explore the surrounding text to figure it out.

I'd strongly recommend to reuse `<fieldset>` plus `<legend>` for all the controls similar to Radius like Padding, Margin, Border but I see there's a lot of inconsistency in regards to the markup used by these controls. It would be great to make them more consistent but that's definitely a topic for another issue.

**Review by ciampo — APPROVED** [review]
Code changes LGTM and test well as per instructions 🚀 

Thank you for working on this improvement, @afercia , and for taking the time to give an overview of how `label`, `aria-label`, tooltips, `fieldset` and `legend` all affect the accessibility tree in different ways 🙇
